src/middlewares.js

```diff
@@ -17,13 +17,15 @@ function handleParseHeaders(req, res, next) {
   var mountPath = req.originalUrl.slice(0, mountPathLength);
   var mount = req.protocol + '://' + req.get('host') + mountPath;
 
+  var basicAuth = httpAuth(req);
+
   var info = {
-    appId: req.get('X-Parse-Application-Id'),
+    appId: basicAuth.appId || req.get('X-Parse-Application-Id'),
```

Comment:
not sure I like mixing those two code paths together. 
Maybe httpAuth should not return anything if nothing is set.

This path may be prone to mix sources from auth and may introduce unintended side effects.